stkSWIV.sol

```diff
@@ -6,59 +6,113 @@ import './Utils/SafeTransferLib.sol';
 
 contract stkSWIV is ERC20 {
 
-    ERC20 immutable SWIV;
+    ERC20 public immutable SWIV;
 
     uint256 public totalDeposited;
 
     constructor (ERC20 s) ERC20("Staked SWIV", "stkSWIV", 18) {
         SWIV = s;
     }
 
+    ////////////////// READ METHODS //////////////////
     function exchangeRateCurrent() public view returns (uint256) {
         return ((SWIV.balanceOf(address(this))*1e26)/totalDeposited);
     }
 
-    function mint(uint256 amount) public returns (uint256) {
+    function totalAssets() public view returns (uint256 totalManagedAssets) {
+        return (SWIV.balanceOf(address(this)));
+    }
+
+    function asset() public view returns (ERC20 assetTokenAddress) {
+        return(SWIV);
+    }
```

Comment:
According to the `ERC-4626` EIP, this method should return an `address` type. I also think returning an address type would be easier if `SWIV` is stored as an `address` instead of an `ERC20`.